pkg/upgrade/upgrader/jiva_upgrade.go

```diff
@@ -633,3 +683,225 @@ func jivaUpgrade(pvName, openebsNamespace string, utaskObj *utask.UpgradeTask) (
 	klog.Info("Upgrade Successful for", pvName)
 	return options.utaskObj, nil
 }
+
+func (j *jivaVolumeOptions) cleanup(openebsNamespace string) error {
+	var err error
+	if j.replicaObj.version == currentVersion && j.replicaObj.name != "" {
+		klog.Info("cleaning old replica deployment")
+		err = deployClient.WithNamespace(j.ns).Delete(j.replicaObj.name, &metav1.DeleteOptions{})
+		if err != nil {
+			return err
+		}
+	}
+	if j.controllerObj.version == currentVersion && j.ns != openebsNamespace {
+		klog.Info("cleaning old replica deployment")
```

Comment:
n_: this should be deleting controller(or target) deployment.

Reply:
Done